`azd ext source remove` with no arguments panics

```
$ azd ext source remove

Remove extension source (azd extension source remove)

panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/azure/azure-dev/cli/azd/cmd.(*extensionSourceRemoveAction).Run(0xc000296780, {0x1725488, 0xc000296240})
        /workspaces/azure-dev/cli/azd/cmd/extension.go:861 +0x37a
github.com/azure/azure-dev/cli/azd/cmd/middleware.(*MiddlewareRunner).RunAction.func1({0x1725488, 0xc000296240})
        /workspaces/azure-dev/cli/azd/cmd/middleware/middleware.go:129 +0x429
github.com/azure/azure-dev/cli/azd/cmd/middleware.(*TelemetryMiddleware).Run(0xc00028a0f0, {0x1725488, 0xc00024f9e0}, 0xc000256c00)
        /workspaces/azure-dev/cli/azd/cmd/middleware/telemetry.go:81 +0x8e6
github.com/azure/azure-dev/cli/azd/cmd/middleware.(*MiddlewareRunner).RunAction.func1({0x1725488, 0xc00024f9e0})
        /workspaces/azure-dev/cli/azd/cmd/middleware/middleware.go:112 +0x2b8
github.com/azure/azure-dev/cli/azd/cmd/middleware.(*UxMiddleware).Run(0xc000280060, {0x1725488, 0xc00024f9e0}, 0xc000256c00)
        /workspaces/azure-dev/cli/azd/cmd/middleware/ux.go:35 +0xa5
github.com/azure/azure-dev/cli/azd/cmd/middleware.(*MiddlewareRunner).RunAction.func1({0x1725488, 0xc00024f9e0})
        /workspaces/azure-dev/cli/azd/cmd/middleware/middleware.go:112 +0x2b8
github.com/azure/azure-dev/cli/azd/cmd/middleware.(*DebugMiddleware).Run(0xc000280030, {0x1725488, 0xc00024f9e0}, 0xc000256c00)
        /workspaces/azure-dev/cli/azd/cmd/middleware/debug.go:51 +0x382
github.com/azure/azure-dev/cli/azd/cmd/middleware.(*MiddlewareRunner).RunAction.func1({0x1725488, 0xc00024f9e0})
        /workspaces/azure-dev/cli/azd/cmd/middleware/middleware.go:112 +0x2b8
github.com/azure/azure-dev/cli/azd/cmd/middleware.(*MiddlewareRunner).RunAction(0xc000260a40, {0x1725488, 0xc00024f9e0}, 0xc00041af00, {0xc000262810, 0x22})
        /workspaces/azure-dev/cli/azd/cmd/middleware/middleware.go:133 +0x137
github.com/azure/azure-dev/cli/azd/cmd.(*CobraBuilder).configureActionResolver.func1(0xc0003d4b08, {0x2091b20, 0x0, 0x0})
        /workspaces/azure-dev/cli/azd/cmd/cobra_builder.go:133 +0x37c
github.com/spf13/cobra.(*Command).execute(0xc0003d4b08, {0x2091b20, 0x0, 0x0})
        /go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:856 +0x68a
github.com/spf13/cobra.(*Command).ExecuteC(0xc0004e6848)
        /go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:974 +0x38d
github.com/spf13/cobra.(*Command).Execute(...)
        /go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:902
github.com/spf13/cobra.(*Command).ExecuteContext(...)
        /go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:895
main.main()
        /workspaces/azure-dev/cli/azd/main.go:65 +0x24b
```

@wbreza FYI

--- a/cli/azd/cmd/extension.go
+++ b/cli/azd/cmd/extension.go
@@ -49,7 +49,6 @@ func extensionActions(root *actions.ActionDescriptor) *actions.ActionDescriptor 
 		Command: &cobra.Command{
 			Use:   "show <extension-name>",
 			Short: "Show details for a specific extension.",
-			Args:  cobra.ExactArgs(1),
 		},
 		OutputFormats:  []output.Format{output.JsonFormat, output.NoneFormat},
 		DefaultFormat:  output.NoneFormat,
@@ -364,6 +363,12 @@ func (t *extensionShowItem) Display(writer io.Writer) error {
 }
 
 func (a *extensionShowAction) Run(ctx context.Context) (*actions.ActionResult, error) {
+	if len(a.args) == 0 {
+		return nil, fmt.Errorf("must specify an extension name")
+	}
+	if len(a.args) > 1 {
+		return nil, fmt.Errorf("cannot specify multiple extensions")
+	}
 	extensionId := a.args[0]
 	filterOptions := &extensions.FilterOptions{
 		Source: a.flags.source,
@@ -873,6 +878,12 @@ func newExtensionSourceRemoveAction(
 }
 
 func (a *extensionSourceRemoveAction) Run(ctx context.Context) (*actions.ActionResult, error) {
+	if len(a.args) == 0 {
+		return nil, fmt.Errorf("must specify an extension source name")
+	}
+	if len(a.args) > 1 {
+		return nil, fmt.Errorf("cannot specify multiple extension sources")
+	}
 	a.console.MessageUxItem(ctx, &ux.MessageTitle{
 		Title: "Remove extension source (azd extension source remove)",
 	})

--- a/cli/azd/cmd/extension.go
+++ b/cli/azd/cmd/extension.go
@@ -172,12 +172,12 @@ func newExtensionListAction(
 }
 
 type extensionListItem struct {
-	Id        string `json:"id"`
-	Name      string `json:"name"`
-	Namespace string `json:"namespace"`
-	Version   string `json:"version"`
-	Installed bool   `json:"installed"`
-	Source    string `json:"source"`
+	Id               string `json:"id"`
+	Name             string `json:"name"`
+	Namespace        string `json:"namespace"`
+	Version          string `json:"version"`
+	InstalledVersion string `json:"installedVersion"`
+	Source           string `json:"source"`
 }
 
 func (a *extensionListAction) Run(ctx context.Context) (*actions.ActionResult, error) {
@@ -212,20 +212,18 @@ func (a *extensionListAction) Run(ctx context.Context) (*actions.ActionResult, e
 			continue
 		}
 
-		var version string
-		if has {
-			version = installedExtension.Version
-		} else {
-			version = extension.Versions[len(extension.Versions)-1].Version
+		var installedVersion string
+		if installed {
+			installedVersion = installedExtension.Version
 		}
 
 		extensionRows = append(extensionRows, extensionListItem{
-			Id:        extension.Id,
-			Name:      extension.DisplayName,
-			Namespace: extension.Namespace,
-			Version:   version,
-			Source:    extension.Source,
-			Installed: installed,
+			Id:               extension.Id,
+			Name:             extension.DisplayName,
+			Namespace:        extension.Namespace,
+			Version:          extension.Versions[len(extension.Versions)-1].Version,
+			InstalledVersion: installedVersion,
+			Source:           extension.Source,
 		})
 	}
 
@@ -264,12 +262,12 @@ func (a *extensionListAction) Run(ctx context.Context) (*actions.ActionResult, e
 				ValueTemplate: `{{.Version}}`,
 			},
 			{
-				Heading:       "Source",
-				ValueTemplate: `{{.Source}}`,
+				Heading:       "Installed Version",
+				ValueTemplate: `{{.InstalledVersion}}`,
 			},
 			{
-				Heading:       "Installed",
-				ValueTemplate: `{{.Installed}}`,
+				Heading:       "Source",
+				ValueTemplate: `{{.Source}}`,
 			},
 		}
 

--- a/cli/azd/cmd/extension.go
+++ b/cli/azd/cmd/extension.go
@@ -54,6 +54,7 @@ func extensionActions(root *actions.ActionDescriptor) *actions.ActionDescriptor 
 		OutputFormats:  []output.Format{output.JsonFormat, output.NoneFormat},
 		DefaultFormat:  output.NoneFormat,
 		ActionResolver: newExtensionShowAction,
+		FlagsResolver:  newExtensionShowFlags,
 	})
 
 	// azd extension install <extension-name>
@@ -283,8 +284,19 @@ func (a *extensionListAction) Run(ctx context.Context) (*actions.ActionResult, e
 }
 
 // azd extension show
+type extensionShowFlags struct {
+	source string
+}
+
+func newExtensionShowFlags(cmd *cobra.Command) *extensionShowFlags {
+	flags := &extensionShowFlags{}
+	cmd.Flags().StringVarP(&flags.source, "source", "s", "", "The extension source to use.")
+	return flags
+}
+
 type extensionShowAction struct {
 	args             []string
+	flags            *extensionShowFlags
 	formatter        output.Formatter
 	writer           io.Writer
 	extensionManager *extensions.Manager
@@ -292,12 +304,14 @@ type extensionShowAction struct {
 
 func newExtensionShowAction(
 	args []string,
+	flags *extensionShowFlags,
 	formatter output.Formatter,
 	writer io.Writer,
 	extensionManager *extensions.Manager,
 ) actions.Action {
 	return &extensionShowAction{
 		args:             args,
+		flags:            flags,
 		formatter:        formatter,
 		writer:           writer,
 		extensionManager: extensionManager,
@@ -306,6 +320,7 @@ func newExtensionShowAction(
 
 type extensionShowItem struct {
 	Id               string
+	Source           string
 	Namespace        string
 	Description      string
 	Tags             []string
@@ -325,6 +340,7 @@ func (t *extensionShowItem) Display(writer io.Writer) error {
 		output.TableFlags)
 	text := [][]string{
 		{"Id", ":", t.Id},
+		{"Source", ":", t.Source},
 		{"Namespace", ":", t.Namespace},
 		{"Description", ":", t.Description},
 		{"Latest Version", ":", t.LatestVersion},
@@ -351,7 +367,11 @@ func (t *extensionShowItem) Display(writer io.Writer) error {
 
 func (a *extensionShowAction) Run(ctx context.Context) (*actions.ActionResult, error) {
 	extensionId := a.args[0]
-	registryExtension, err := a.extensionManager.GetFromRegistry(ctx, extensionId, nil)
+	filterOptions := &extensions.FilterOptions{
+		Source: a.flags.source,
+	}
+
+	registryExtension, err := a.extensionManager.GetFromRegistry(ctx, extensionId, filterOptions)
 	if err != nil {
 		return nil, fmt.Errorf("failed to get extension details: %w", err)
 	}
@@ -360,6 +380,7 @@ func (a *extensionShowAction) Run(ctx context.Context) (*actions.ActionResult, e
 
 	extensionDetails := extensionShowItem{
 		Id:               registryExtension.Id,
+		Source:           registryExtension.Source,
 		Namespace:        registryExtension.Namespace,
 		Description:      registryExtension.DisplayName,
 		Tags:             registryExtension.Tags,
@@ -372,7 +393,7 @@ func (a *extensionShowAction) Run(ctx context.Context) (*actions.ActionResult, e
 	installedExtension, err := a.extensionManager.GetInstalled(
 		extensions.LookupOptions{Id: extensionId},
 	)
-	if err == nil {
+	if err == nil && installedExtension.Source == extensionDetails.Source {
 		extensionDetails.InstalledVersion = installedExtension.Version
 	}
 

--- a/cli/azd/docs/extension-framework.md
+++ b/cli/azd/docs/extension-framework.md
@@ -44,6 +44,12 @@ The official extension source registry is pre-configured in `azd` and is hosted 
 
 The registry is hosted in the [`azd` github repo](https://github.com/Azure/azure-dev/blob/main/cli/azd/extensions/registry.json).
 
+If you previously removed it and want to add it back:
+
+```bash
+azd extension source add -n azd -t url -l "https://aka.ms/azd/extensions/registry"
+```
+
 #### Development Registry
 
 > [!CAUTION]
@@ -81,11 +87,17 @@ Extensions are a collection of executable artifacts that extend or enhance funct
 
 #### `azd extension list [flags]`
 
-Lists matching extensions from one or more extension sources
+Lists matching extensions from one or more extension sources.
 
 - `--installed` When set displays a list of installed extensions.
 - `--source` When set will only list extensions from the specified source.
 - `--tags` Allows filtering extensions by tags (e.g., AI, test)
+
+#### `azd extension show <extension-name> [flags]`
+
+Shows detailed information for a specific extension, including description, tags, versions, and installation status.
+
+- `-s, --source` The extension source to use. Use this flag when the same extension ID exists in multiple sources.
 
 #### `azd extension install <extension-names> [flags]`
 

--- a/cli/azd/pkg/extensions/manager.go
+++ b/cli/azd/pkg/extensions/manager.go
@@ -24,6 +24,7 @@ import (
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
 	azruntime "github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
 	"github.com/Masterminds/semver/v3"
+	"github.com/azure/azure-dev/cli/azd/internal"
 	"github.com/azure/azure-dev/cli/azd/pkg/alpha"
 	"github.com/azure/azure-dev/cli/azd/pkg/config"
 	"github.com/azure/azure-dev/cli/azd/pkg/osutil"
@@ -158,7 +159,8 @@ func (m *Manager) GetInstalled(options LookupOptions) (*Extension, error) {
 	return nil, ErrInstalledExtensionNotFound
 }
 
-// GetFromRegistry retrieves an extension from the registry by name
+// GetFromRegistry retrieves an extension from the registry by name.
+// Returns an error if multiple extensions with the same ID are found across different sources.
 func (m *Manager) GetFromRegistry(
 	ctx context.Context,
 	extensionId string,
@@ -181,7 +183,7 @@ func (m *Manager) GetFromRegistry(
 		return nil, fmt.Errorf("failed getting extension sources: %w", err)
 	}
 
-	var match *ExtensionMetadata
+	var matches []*ExtensionMetadata
 	var sourceErr error
 
 	for _, source := range sources {
@@ -189,13 +191,23 @@ func (m *Manager) GetFromRegistry(
 		if err != nil {
 			sourceErr = err
 		} else if extension != nil {
-			match = extension
-			break
+			matches = append(matches, extension)
 		}
 	}
 
-	if match != nil {
-		return match, nil
+	if len(matches) > 1 {
+		sourceNames := make([]string, len(matches))
+		for i, match := range matches {
+			sourceNames[i] = match.Source
+		}
+		return nil, &internal.ErrorWithSuggestion{
+			Err:        fmt.Errorf("extension '%s' found in multiple sources: %s", extensionId, strings.Join(sourceNames, ", ")),
+			Suggestion: "Suggestion: specify the exact source using --source or -s.",
+		}
+	}
+
+	if len(matches) == 1 {
+		return matches[0], nil
 	}
 
 	if sourceErr != nil {

--- a/cli/azd/pkg/extensions/manager.go
+++ b/cli/azd/pkg/extensions/manager.go
@@ -201,7 +201,11 @@ func (m *Manager) GetFromRegistry(
 			sourceNames[i] = match.Source
 		}
 		return nil, &internal.ErrorWithSuggestion{
-			Err:        fmt.Errorf("extension '%s' found in multiple sources: %s", extensionId, strings.Join(sourceNames, ", ")),
+			Err: fmt.Errorf(
+				"extension '%s' found in multiple sources: %s",
+				extensionId,
+				strings.Join(sourceNames, ", "),
+			),
 			Suggestion: "Suggestion: specify the exact source using --source or -s.",
 		}
 	}

--- a/cli/azd/pkg/extensions/manager_test.go
+++ b/cli/azd/pkg/extensions/manager_test.go
@@ -517,3 +517,98 @@ func Test_FindArtifactForCurrentOS_ErrorMessage_Format(t *testing.T) {
 	require.Nil(t, artifact)
 	require.Contains(t, err.Error(), "no artifact available for platform:")
 }
+
+func Test_GetFromRegistry_MultipleMatches_ErrorWithSuggestion(t *testing.T) {
+	mockContext := mocks.NewMockContext(context.Background())
+
+	userConfigManager := config.NewUserConfigManager(mockContext.ConfigManager)
+	sourceManager := NewSourceManager(mockContext.Container, userConfigManager, mockContext.HttpClient)
+
+	// Mock two different registries that both contain the same extension ID
+	mockContext.HttpClient.When(func(request *http.Request) bool {
+		return request.URL.String() == extensionRegistryUrl
+	}).RespondFn(func(request *http.Request) (*http.Response, error) {
+		// Return a registry with an extension
+		registry := Registry{
+			Extensions: []*ExtensionMetadata{
+				{
+					Id:          "duplicate.extension",
+					Namespace:   "duplicate",
+					DisplayName: "Duplicate Extension from Registry",
+					Description: "Extension from first source",
+					Source:      "default", // This will be overwritten by the source
+					Tags:        []string{"test"},
+					Versions: []ExtensionVersion{
+						{
+							Version:   "1.0.0",
+							Artifacts: sampleArtifacts,
+						},
+					},
+				},
+			},
+		}
+		return mocks.CreateHttpResponseWithBody(request, http.StatusOK, registry)
+	})
+
+	// Create a manager and mock two sources
+	manager, err := NewManager(userConfigManager, sourceManager, mockContext.HttpClient)
+	require.NoError(t, err)
+
+	// Create mock sources that will return the same extension
+	mockSource1 := &mockSource{
+		name: "source1",
+		extensions: []*ExtensionMetadata{
+			{
+				Id:          "duplicate.extension",
+				Namespace:   "duplicate",
+				DisplayName: "Extension from Source 1",
+				Source:      "source1",
+			},
+		},
+	}
+
+	mockSource2 := &mockSource{
+		name: "source2",
+		extensions: []*ExtensionMetadata{
+			{
+				Id:          "duplicate.extension",
+				Namespace:   "duplicate",
+				DisplayName: "Extension from Source 2",
+				Source:      "source2",
+			},
+		},
+	}
+
+	// Override the sources with our mocks
+	manager.sources = []Source{mockSource1, mockSource2}
+
+	// Try to get the extension - should return error with suggestion
+	extension, err := manager.GetFromRegistry(*mockContext.Context, "duplicate.extension", nil)
+
+	// Verify we got the expected error
+	require.Error(t, err)
+	require.Nil(t, extension)
+}
+
+// mockSource is a test implementation of the Source interface
+type mockSource struct {
+	name       string
+	extensions []*ExtensionMetadata
+}
+
+func (m *mockSource) Name() string {
+	return m.name
+}
+
+func (m *mockSource) ListExtensions(ctx context.Context) ([]*ExtensionMetadata, error) {
+	return m.extensions, nil
+}
+
+func (m *mockSource) GetExtension(ctx context.Context, extensionId string) (*ExtensionMetadata, error) {
+	for _, ext := range m.extensions {
+		if ext.Id == extensionId {
+			return ext, nil
+		}
+	}
+	return nil, nil
+}

--- a/cli/azd/pkg/extensions/source_manager.go
+++ b/cli/azd/pkg/extensions/source_manager.go
@@ -8,7 +8,7 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
-	"sort"
+	"slices"
 	"strings"
 
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
@@ -167,8 +167,8 @@ func (sm *SourceManager) List(ctx context.Context) ([]*SourceConfig, error) {
 		allSourceConfigs = append(allSourceConfigs, defaultSource)
 	}
 
-	sort.Slice(allSourceConfigs, func(i, j int) bool {
-		return allSourceConfigs[i].Name < allSourceConfigs[j].Name
+	slices.SortFunc(allSourceConfigs, func(a, b *SourceConfig) int {
+		return strings.Compare(a.Name, b.Name)
 	})
 
 	return allSourceConfigs, nil

--- a/cli/azd/pkg/extensions/source_manager.go
+++ b/cli/azd/pkg/extensions/source_manager.go
@@ -8,6 +8,7 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
+	"sort"
 	"strings"
 
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
@@ -165,6 +166,10 @@ func (sm *SourceManager) List(ctx context.Context) ([]*SourceConfig, error) {
 
 		allSourceConfigs = append(allSourceConfigs, defaultSource)
 	}
+
+	sort.Slice(allSourceConfigs, func(i, j int) bool {
+		return allSourceConfigs[i].Name < allSourceConfigs[j].Name
+	})
 
 	return allSourceConfigs, nil
 }

